File: code/jvm/src/main/kotlin/pt/isel/daw/battleships/controllers/games/models/ship/ShipTypeModel.kt
package pt.isel.daw.battleships.controllers.games.models.ship

import pt.isel.daw.battleships.services.games.dtos.ship.ShipTypeDTO

/**
 * Represents a ship type.
 *
 * @property shipName The name of the ship type.
 * @property size The size of the ship type.
 * @property points The points of the ship type.
 */
data class ShipTypeModel(
    val shipName: String,
    val size: Int,
    val quantity: Int,
    val points: Int
) {
    fun toShipTypeDTO() = ShipTypeDTO(shipName, size, quantity, points)

    constructor(shipType: ShipTypeDTO) : this(
        shipType.shipName,
        shipType.size,
        shipType.quantity,
        shipType.points
    )
}

File: code/js/src/services/games/models/ShipTypeModel.ts
/**
 * The Ship Type Model.
 *
 * @property shipName the name of the ship
 * @property size the size of the ship
 * @property quantity the quantity of ships of this type
 * @property points the points that the ship is worth
 */
interface ShipTypeModel {
    shipName: string;
    size: number;
    quantity: number;
    points: number;
}
